hermes/serializers.py

```diff
@@ -224,25 +411,66 @@ def validate_dec(self, value):
         return dec_angle.deg
 
 
-class PhotometryDataSerializer(serializers.Serializer):
-    event_id = serializers.CharField(required=False)
+class GenericHermesDataSerializer(serializers.Serializer):
+    references = ReferenceDataSerializer(many=True, required=False)
     extra_data = serializers.JSONField(required=False)
-    photometry = PhotometrySerializer(many=True)
-
-    def validate_photometry(self, value):
-        if len(value) < 1:
-            raise serializers.ValidationError(_('At least one piece of photometry must be defined'))
-        return value
-
-
-class HermesPhotometrySerializer(HermesMessageSerializer):
-    data = PhotometryDataSerializer()
+    event_id = serializers.CharField(required=False)
+    targets = TargetDataSerializer(many=True, required=False)
+    photometry = PhotometryDataSerializer(many=True, required=False)
+    spectroscopy = SpectroscopyDataSerialzier(many=True, required=False)
+    astrometry = AstrometryDataSerializer(many=True, required=False)
 
+    def validate(self, data):
```

Comment:
Should we also validate that every target in the target list is referenced in either another table or the message text somehow? Presumably we don't want people to just submit a massive target list every time with a bunch of irrelevant targets?